Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -0,0 +1,240 @@
+
+        for position in positions {
+            validateAnnotation(byAnnotationId: position.identifier)
+            guard let annotationView = annotationViewsById[position.identifier] else {
+                continue
+            }
+            annotationView.translatesAutoresizingMaskIntoConstraints = true
+            annotationView.frame = CGRect(
+                origin: position.leftTopCoordinate.point,
+                size: CGSize(width: CGFloat(position.width), height: CGFloat(position.height))
+            )
+            annotationView.setInternalVisibility(isHidden: false)
+            visibleAnnotationIds.insert(position.identifier)
+        }
+
+        let annotationsToHide = Set<String>(annotationViewsById.keys).subtracting(visibleAnnotationIds)
+        for id in annotationsToHide {
+            validateAnnotation(byAnnotationId: id)
+            if let annotationView = annotationViewsById[id] {
+                annotationView.setInternalVisibility(isHidden: true)
+            }
+        }
+    }
+
+    internal func validateAnnotation(byAnnotationId id: String) {
+        guard let annotationView = annotationViewsById[id] else { return }
+        // If the user explicitly removed the ViewAnnotation or it's wrapped view
+        // we need to remove it from our layout calculation
+        if annotationView.subviews.isEmpty || annotationView.superview == nil {
+            try? remove(annotationView)
+        }
+        // View is still considered for layout calculation, users should not modify the visibility of the wrapped view
+        if let wrappedView = annotationView.subviews.first, wrappedView.isHidden {
+            Log.warning(forMessage: "Visibility changed for wrapped view", category: "Annotations")
+        }
+    }
+
+}
+
+extension ViewAnnotationManager: DelegatingViewAnnotationPositionsUpdateListenerDelegate {
+
+    internal func onViewAnnotationPositionsUpdate(forPositions positions: [ViewAnnotationPositionDescriptor]) {
+        placeAnnotations(positions: positions)
+    }
+
+}
+
+public final class AnnotationView: SubviewInteractionOnlyView {
+
+    private static var currentId = 0
+    internal let id: String = {
+        let id = String(currentId)
+        currentId += 1
+        return id
+    }()
+    internal let wrappedView: UIView
+    internal var ignoreUserEvents: Bool = false
+
+    // In case the user changes the visibility of the AnnotationView, we should update the options to remove it from the layout calculation
+    public override var isHidden: Bool {
+        didSet {
+            guard !ignoreUserEvents else { return }
+            guard let manager = annotationManager else { return }
+            let options = manager.options(byAnnotationView: self)
+            let visible = !isHidden
+            if visible != options?.visible {
+                try? manager.update(self, options: ViewAnnotationOptions(visible: visible))
+            }
+        }
+    }
```

Comment:
This seems like the only behavior that warrants exposing `AnnotationView` as a public API. I am concerned that making `AnnotationView` public is too limiting to future iteration that we might want to do on this implementation.

Could we do some more exploration to see if there's another reasonable way to work around this issue? (and even confirm that it's worth working around given the complexity & maintenance commitment)?

Reply:
One alternative I was considering is just to show a warning, like it's already implemented for the subview's visibility state in `validateAnnotation`.
The only issue with that is the user won't see it until they move the map around, and check the logs for warnings.
And we can also highlight in the docs that hiding the view won't remove them from the layout calculation.